### PR TITLE
Feature/798 Add force unaggregate to schedule A

### PR DESCRIPTION
**Pull request by mjtravers**
#798

### CODE REVIEW

**Review by toddlees — CHANGES_REQUESTED** [review]
we calculate a preview of the aggregate on the transaction form. is it supposed to not factor in the contribution amount if the transaction is unaggregated.  e.g. if the previous transaction aggregated to $200, and the user is editing a $100, unaggregated transaction, should the aggregate show $200?

**Comment by mjtravers** [comment]
> we calculate a preview of the aggregate on the transaction form. is it supposed to not factor in the contribution amount if the transaction is unaggregated. e.g. if the previous transaction aggregated to $200, and the user is editing a $100, unaggregated transaction, should the aggregate show $200?

Ah, I will correct that. No, I don't think the JIT update in the UI should update the aggregate if they've selected force unaggregate. I will correct that.

Confirmed by Gregg: If user has manually aggregated or unaggregated a transaction then the system should not dynamically show something different.

File: front-end/src/app/shared/components/transaction-type-base/transaction-form.utils.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class TransactionFormUtils {
         });
     }
 
-    if (transactionType.showAggregate) {
+    if (transactionType.showAggregate && !transaction.force_unaggregated) {
```
Comment (toddlees) [inline front-end/src/app/shared/components/transaction-type-base/transaction-form.utils.ts:92]: looks like the aggregate still needs to show for these transactions.  the difference being that we don't add the amount of the unaggegated transaction to the aggregate.  seems like we could update `updateAggregate` to add $0 if the transaction is unaggregated

**Review by toddlees — CHANGES_REQUESTED** [review]

File: front-end/src/app/shared/components/transaction-type-base/transaction-form.utils.ts (hunk 1 above)
Comment (mjtravers) [reply front-end/src/app/shared/components/transaction-type-base/transaction-form.utils.ts:92]: I believe that is how it is working currently. The !transaction.force_unaggregated above simply keeps values and changes made to the Amount in the form from adding to the displayed Aggregate, which is the value from the previous transaction.

**Review by toddlees — APPROVED** [review]

**Pull request merged**